src/validators/SessionKeyValidator.sol

```diff
@@ -123,41 +124,56 @@ contract SessionKeyValidator is IValidationHook, IModuleValidator {
    * @return true if validator is registered for the account, false otherwise
    */
   function isInitialized(address smartAccount) external view returns (bool) {
-    return _isHookAndModuleInitialized(smartAccount);
+    return _isInitialized(smartAccount);
   }
 
-  function _isHookAndModuleInitialized(address smartAccount) internal view returns (bool) {
-    return _isHookInitialized(smartAccount) && _isModuleInitialized(smartAccount);
-  }
-
-  function _isHookInitialized(address smartAccount) internal view returns (bool) {
-    return IHookManager(smartAccount).isHook(address(this));
+  function _isInitialized(address smartAccount) internal view returns (bool) {
+    return IValidatorManager(smartAccount).isModuleValidator(address(this));
   }
 
   function _isModuleInitialized(address smartAccount) internal view returns (bool) {
     return IValidatorManager(smartAccount).isModuleValidator(address(this));
   }
 
-  function validationHook(bytes32 signedHash, Transaction calldata transaction, bytes calldata hookData) external {
-    (bytes memory signature, address validator, ) = abi.decode(transaction.signature, (bytes, address, bytes[]));
+  // this generally throws instead of returning false
+  function _isValidTransaction(
+    bytes32 signedHash,
+    bytes memory _signature,
+    Transaction calldata transaction
+  ) internal returns (bool) {
+    (bytes memory transactionSignature, address validator, bytes[] memory moduleData) = abi.decode(
+      transaction.signature,
+      (bytes, address, bytes[])
+    );
     if (validator != address(this)) {
       // This transaction is not meant to be validated by this module
-      return;
+      return false;
     }
+
     (SessionLib.SessionSpec memory spec, uint64[] memory periodIds) = abi.decode(
-      hookData,
+      moduleData[0], // this is known by the signature builder
       (SessionLib.SessionSpec, uint64[])
     );
-    (address recoveredAddress, ) = ECDSA.tryRecover(signedHash, signature);
-    require(recoveredAddress == spec.signer, "Invalid signer");
+    require(spec.signer != address(0), "Invalid signer (empty)");
+    (address recoveredAddress, ECDSA.RecoverError recoverError) = ECDSA.tryRecover(signedHash, transactionSignature);
+
+    // gas estimation provides invalid custom signatures
+    if (recoveredAddress == address(0) && recoverError == ECDSA.RecoverError.InvalidSignature) {
+      return false;
```

Comment:
I initially thought this was necessary due to gas estimation (as the comment implies), but nothing should have changed in the gas estimation flow for these transactions so this should have failed before. The signing error indicates that the transaction signatures are different (without this block it fails with "Invalid signer (mismatch), so it's not very clear where this other signature comes from if not supplied during gas estimation.

This being an early-out for gas estimation only I tried to increase the amount of gas used in this path, but it also increased it for the actual transaction. Here's a quick table of results by running the tests against both.

| return false  | use extra gas during this branch |
| ------------- | ------------- |
| `sessionTx` gas estimated: 346191  | `sessionTx` gas estimated: 333947 |
| `sessionTx` gas used: 266405  | `sessionTx` gas used: 266457  |
| `sessionTx` gas estimated: 346191 | `sessionTx` gas estimated: 346986 |
| `sessionTx` gas used: 311767 | `sessionTx` gas used: 311819 |
| `sessionTx` gas estimated: 332361 | `sessionTx` gas estimated: 333947 |
| `sessionTx` gas used: 309578 | `sessionTx` gas used: 309630  |
| `sessionTx` gas estimated: 272009 | `sessionTx` gas estimated: 273595 |
| `sessionTx` gas used: 263170 | `sessionTx` gas used: 263222 |